server/s3_bucket_metadata.h

```diff
@@ -56,6 +56,7 @@ class S3BucketMetadata {
   // http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html
   std::string bucket_name;
   std::string bucket_policy;
+  std::string bucket_versioning_status{"Unversioned"};
```

Comment:
Should we enumerate the status? It will be used in many places to check bucket status.

Reply:
As discuss with team, its fine to have it as a string.